extensions/opentelemetry/opentelemetry-async-mutiny/runtime/src/main/java/io/quarkus/opentelemetry/async/mutiny/runtime/tracing/OpenTelemetryMultiInterceptor.java

```diff
@@ -0,0 +1,28 @@
+package io.quarkus.opentelemetry.async.mutiny.runtime.tracing;
+
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+
+import io.opentelemetry.context.Context;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.infrastructure.MultiInterceptor;
+
+public class OpenTelemetryMultiInterceptor implements MultiInterceptor {
```

Comment:
Unfortunately, this will have a massive impact on the performance. I don't know a better way, but that's something to keep in mind.

Reply:
I know. When I check other implementation like rxjava it seems like this is necessary:
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/rxjava/rxjava-2.0/library/src/main/java/io/opentelemetry/instrumentation/rxjava/v2_0/TracingAssembly.java

I do not know if there is a different way to achieve tracing in reactive. But I am afraid it is not. This is why I hoped that the Munity guys say: No, no Chris there is a way easier approach to achieve the same. Without that I always ended up with broken, not connected traces, which is not really helpful.